### PR TITLE
Add isModified model method

### PR DESCRIPTION
**Pull request by pierreis**
The goal is to mimic the behavior of the `isModified()` model method of Mongoose [1].
Such method can be used, for example to perform expensive validation in pre_save hooks without having to revalidate non-modified fields.

Tests have been updated accordingly.

[1] [http://mongoosejs.com/docs/3.0.x/docs/api.html#document_Document-isModified](http://mongoosejs.com/docs/3.0.x/docs/api.html#document_Document-isModified)

### CODE REVIEW

File: src/orm/base_model.js [hunk 1]
```diff
@@ -2385,4 +2404,11 @@ BaseModel.prototype.toJSON = function toJSON() {
   return object;
 };
 
+BaseModel.prototype.isModified = function isModified(propName) {
```
Comment (masumsoft) [inline src/orm/base_model.js]: Could you please add some documentation with a possible use case for this method too?

File: test/test.js [hunk 2]
```diff
@@ -387,6 +397,11 @@ describe('Unit Tests', () => {
         person.intMapDefault.should.deep.equal({ one: 1, two: 2 });
         person.stringListDefault.should.have.members(['one', 'two']);
         person.intSetDefault.should.have.members([1, 2]);
+        person.isModified().should.equal(false);
+        person.Name = 'john';
+        person.isModified('Name').should.equal(true);
```
Comment (masumsoft) [inline test/test.js]: could you please add another check for a non modified valid field name?

**Comment by masumsoft** [comment]
Also could you please address the remaining code climate issues?

**Comment by pierreis** [comment]
I just fixed all issues with the code.

**Comment by masumsoft** [comment]
The merged code is published to npm v1.8.0

**Pull request merged**